pkg/providers/pricing/pricing.go

```diff
@@ -218,7 +230,8 @@ func (p *DefaultProvider) UpdateOnDemandPricing(ctx context.Context) error {
 		return fmt.Errorf("no on-demand pricing found")
 	}
 
-	p.onDemandPrices = lo.Assign(onDemandPrices, onDemandMetalPrices)
+	// Maintain previously retrieved pricing data
+	p.onDemandPrices = lo.Assign(p.onDemandPrices, onDemandPrices, onDemandMetalPrices)
```

Comment:
I think this was fine the way it was right since we're not doing this assignment in the paginator like we were for spot?

Reply:
The issues was more that we were fully overriding the pricing data -- I think we want to retain the existing data if we don't have replacements for it -- imagine we get some bad response from the pricing API due to some issue with it -- I think we want to try to hold onto the data if we can.